### ISSUE TITLE
Rule Change: Introduce `func-style` `allowAnonymousFunctions` option

### ISSUE THREAD

**Issue by mjfaga** [eslint/eslint#17128]
### What rule do you want to change?

func-style

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
## Proposed option
* `allowAnonymousFunctions` - when set, anonymous functions can be used following declaration conventions by storing in a `const`
  
          "func-style": ["error", "declaration", {allowAnonymousFunctions: true}],
  

## Currently, this fails
```typescript
const myFunction: SomeFunctionSignature = function (...args) {
  // do something
};
```
```


### What does the rule currently do for this code?

When `style='declaration'`, this code gets a warning.

### What will the rule do after it's changed?

When the `allowAnonymousFunctions` is set to `true`, the rule will not warn on this.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

_No response_

**Comment by Rec0iL99** [comment]
Hi @mjfaga, thanks for the issue. As mentioned in the [README](https://github.com/eslint/eslint#stylistic-rule-updates), we currently aren't adding any new options to stylistic rules in ESLint. You can always copy the core rule and add options to suit your needs.
